Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -462,20 +453,21 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
         }
     }
 
-    private func sendErrorEvent(on command: RUMAddCurrentViewErrorCommand) -> Bool {
+    private func sendErrorEvent(on command: RUMAddCurrentViewErrorCommand, context: DatadogV1Context, writer: Writer) {
+        errorsCount += 1
         attributes.merge(rumCommandAttributes: command.attributes)
 
-        let eventData = RUMErrorEvent(
+        let errorEvent = RUMErrorEvent(
             dd: .init(
                 browserSdkVersion: nil,
                 session: .init(plan: .plan1)
             ),
-            action: context.activeUserActionID.flatMap { rumUUID in
+            action: self.context.activeUserActionID.map { rumUUID in
                 .init(id: rumUUID.toRUMDataFormat)
             },
-            application: .init(id: context.rumApplicationID),
+            application: .init(id: self.context.rumApplicationID),
             ciTest: dependencies.ciTest,
-            connectivity: dependencies.connectivityInfoProvider.current,
+            connectivity: .init(context: context),
```

Comment:
Nice 👍. I think we should do the same for initializing other RUM data types from `context`, e.g.: `RUMOperatingSystem`, `RUMDevice`. Can be a separate PR IMO.